scripts/zkai_attention_derived_d128_snark_statement_receipt_gate.py

```
@@ -0,0 +1,983 @@
+        "message": "the existing two-slice SNARK receipt public signals do not bind this attention-derived contract",
+    }
+
+
+def run_gate(external_verify: Callable[[dict[str, Any], list[Any], dict[str, Any]], None] = snarkjs_verify) -> dict[str, Any]:
+    baseline = baseline_receipt()
+    verify_proof_only(baseline, external_verify=external_verify)
+    verify_statement_receipt(baseline, external_verify=external_verify)
+    proof_verifier_check = proof_verifier_public_signal_check(baseline, external_verify)
+    mutations = mutated_receipts()
+    if set(mutations) != EXPECTED_MUTATION_SET:
+        raise RuntimeError("mutation corpus does not match expected attention-derived d128 SNARK receipt suite")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)